cli/src/main/scala/aqua/AquaCompiler.scala

```diff
@@ -18,19 +19,30 @@ import cats.syntax.monoid._
 import cats.syntax.show._
 import fs2.io.file.Files
 import fs2.text
+import wvlet.log.LogSupport
 
 import java.nio.file.Path
 
-object AquaCompiler {
+object AquaCompiler extends LogSupport {
   sealed trait CompileTarget
   case object TypescriptTarget extends CompileTarget
   case object AirTarget extends CompileTarget
 
-  case class Prepared(target: String => Path, model: ScriptModel) {
+  case class Prepared(srcFile: Path, srcDir: Path, model: ScriptModel) {
 
     def hasOutput(target: CompileTarget): Boolean = target match {
       case _ => model.funcs.nonEmpty
     }
+
+    def targetPath(ext: String): Validated[Throwable, Path] =
+      Validated.catchNonFatal {
+        val fileName = srcFile.getFileName
+        if (fileName == null) {
+          throw new Exception(s"Unexpected: there is no file name in $srcFile")
```

Comment:
You mean, `$fileName`?

Reply:
it will be null, so, no

Reply:
I don't understand what the error message means. Please rephrase.